### PR TITLE
add child to parent in completion context

### LINKED ISSUE #21154: reproducible builds: undeterministic additional source path references in tasty
We encountered a scala 3 binary reproducibility problem in Pekko (https://github.com/apache/pekko-http/issues/566, https://github.com/apache/pekko-connectors/issues/739): on a clean compile, `FcmOption.tasty` seems to have references to `FcmJsonSupport`, but it does not seem to be deterministic what `@Child` annotation position id those references point to.

When compiling `FcmOption` in isolation, the references to `FcmJsonSupport` do not appear in the `.tasty` file at all, which begs the question whether they should be created at all.

## Compiler version

3.3.3

(3.4.2 also generated the suspicious references, but we haven't tested yet whether they are deterministic there)

## Minimized code

I have a fairly minimized project showing the surprising references to FcmJsonSupport at https://codeberg.org/raboof/scala3-reproduce-21154 

## Output

```scala
  source paths:
     0: 19 [FcmOption.scala]
    69: 27 [FcmJsonSupport.scala]
```

## Expectation

Maybe no reference to `FcmJsonSupport` at all, or at least deterministic ones.

It seems like, when (e.g.) `FcmOptions` is encountered in `FcmJsonSupport`, the `Child` annotation linking `FcmOptions` and `FcmOption` is already created, but still with the `Context` of `FcmJsonSupport` - though neither `FcmOptions` nor `FcmOption` are in that file. It seems at some point the context should be updated, though I'm not familiar enough with the code to make a good suggestion where exactly :)

### PR DESCRIPTION
**Pull request by bishabosha**
This ensures that the positions of forced Child annotations always happens in the source file of the parent class, and not in another file that forced the completion to happen

fixes #21154

### CODE REVIEW

File: tests/pos/i21154/Z.tastycheck [hunk 1]
```diff
@@ -0,0 +1,1340 @@
...
+  1851: 298 .. 298
+  1854: 298 .. 298
+  1859: 298 .. 298
+  1864: 298 .. 298
+  1874: 293 .. 354
+  1880: 293 .. 293
+  1884: 293 .. 293
+
+  source paths:
+     0: 27 [<elided source file name>]
```
Comment (bishabosha) [inline tests/pos/i21154/Z.tastycheck:1337]: the magic is that if you revert the change to Namer, then this will have three source paths - because the positions of the child annotations will be set in the context of `A.scala` - i.e. the call site that forces the completion

**Review by raboof — APPROVED** [review]
I tested this PR with my reproducer at https://codeberg.org/raboof/scala3-reproduce-21154 (invoking scalac directly, no sbt), and it appears to have the intended effect: no more references to `FcmJsonSupport.scala` in `FcmOption.tasty`, and also the previously-nondeterministic 'positions' entries are gone.

**Review by sjrd — APPROVED** [review]
